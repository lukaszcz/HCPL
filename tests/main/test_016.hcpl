
// reduce, eval & quoting test

let f = '\x x;
let g = '\x (f x + 1);
let h = '\x (g x + 1);

print (eval '(f 300));
print (eval '(g 300));
print (eval '(h 300));

print (reduce '(f 300));
print (reduce '(g 300));
print (reduce '(h 300));

symbol VAR;

let v = '(VAR 1);
let k = '\x (x v);

print (k == '\x (x (VAR 1)));
