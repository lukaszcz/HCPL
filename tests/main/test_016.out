300
301
302
300
301
302
true
